maap/config_reader.py

```diff
@@ -1,278 +1,305 @@
+    
+def _get_client_config(maap_host):
+    # This is added to remove the assumption of scheme specially for local dev testing
+    # also maintains backwards compatibility for user to use MAAP("api.maap-project.org")
+    config_url = _get_config_url(maap_host)
+    try:
+        logger.debug(f"Requesting client config from api at: {config_url}")
+        response = requests.get(config_url, verify=False)
+        if response.status_code != 200:
+            raise EnvironmentError("Unable to get config from supplied maap_host")
+        config = response.json()
+        api_root_url = _get_api_root(config_url, config)
+        config["service"]["maap_api_root"] = api_root_url
+        return config
+    except Exception as ex:
+        logger.error(f"Unable to read maap config from api: {ex}")
+
+
+class MaapConfig(metaclass=Singleton):
+    def __init__(self, maap_host=None):
         if maap_host is None:
-            if not self.__config.has_option('service', 'maap_host'):
-                raise KeyError("No maap_host configured in config file or provided as parameter")
-        else:
-            self.__config.set('service', 'maap_host', maap_host)
-        self.__maap_host = self.__config.get("service", "maap_host")
-
-        self.__maap_token = self.__config.get("service", "maap_token")
-        self.__page_size = self.__config.getint("request", "page_size")
-        self.__content_type = self.__config.get("request", "content_type")
-        self.__algorithm_register = self._get_api_endpoint("algorithm_register")
-        self.__algorithm_build = self._get_api_endpoint("algorithm_build")
-        self.__mas_algo = self._get_api_endpoint("mas_algo")
-        self.__dps_job = self._get_api_endpoint("dps_job")
-        self.__wmts = self._get_api_endpoint("wmts")
-        self.__member = self._get_api_endpoint("member")
-        self.__tiler_endpoint = self.__config.get("service", "tiler_endpoint")
-
-
-
-
-        self.__aws_access_key = os.environ.get("AWS_ACCESS_KEY_ID") or self.__config.get("aws", "aws_access_key_id")
-        self.__aws_access_secret = os.environ.get("AWS_SECRET_ACCESS_KEY") or self.__config.get("aws",
-                                                                                              "aws_secret_access_key")
-        self.__s3_user_upload_bucket = os.environ.get("S3_USER_UPLOAD_BUCKET") or self.__config.get("aws",
-                                                                                                  "user_upload_bucket")
-        self.__s3_user_upload_dir = os.environ.get("S3_USER_UPLOAD_DIR") or self.__config.get("aws",
-                                                                                            "user_upload_directory")
-        self.__indexed_attributes = json.loads(self.__config.get("search", "indexed_attributes"))
+            raise EnvironmentError("MAAP is not configured. Please initialize a MAAP object: maap = MAAP()")
```

Comment:
Per my comment on the `_get_client_config` function, you can drop the metaclass (and thus also delete the `Singleton` class), make `maap_host` required, and eliminate the `None` check.

```
class MaapConfig():
    def __init__(self, maap_host):
```

This would also require that `DpsJob.__init__` add a `maap: MAAP` parameter, set `self.maap = maap`, and use `self.maap.config` to obtain the config for the `MAAP` instance rather than calling `MaapConfig()`.